NullPoint

08-23 17:58:14.430 com.lzj.shanyi E/AndroidRuntime: FATAL EXCEPTION: main
                                                    Process: com.lzj.shanyi, PID: 29136
                                                    io.reactivex.exceptions.OnErrorNotImplementedException: Attempt to invoke virtual method 'boolean com.liulishuo.filedownloader.download.DownloadLaunchRunnable.isAlive()' on a null object reference
                                                        at io.reactivex.internal.functions.Functions$14.accept(Functions.java:229)
                                                        at io.reactivex.internal.functions.Functions$14.accept(Functions.java:226)
                                                        at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:72)
                                                        at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:276)
                                                        at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
                                                        at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:252)
                                                        at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:109)
                                                        at android.os.Handler.handleCallback(Handler.java:739)
                                                        at android.os.Handler.dispatchMessage(Handler.java:95)
                                                        at android.os.Looper.loop(Looper.java:152)
                                                        at android.app.ActivityThread.main(ActivityThread.java:5507)
                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                        at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:102)
                                                     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.liulishuo.filedownloader.download.DownloadLaunchRunnable.isAlive()' on a null object reference
                                                        at android.os.Parcel.readException(Parcel.java:1626)
                                                        at android.os.Parcel.readException(Parcel.java:1573)
                                                        at com.liulishuo.filedownloader.i.IFileDownloadIPCService$Stub$Proxy.pause(IFileDownloadIPCService.java:313)
                                                        at com.liulishuo.filedownloader.FileDownloadServiceUIGuard.pause(FileDownloadServiceUIGuard.java:119)
                                                        at com.liulishuo.filedownloader.FileDownloadServiceProxy.pause(FileDownloadServiceProxy.java:72)
                                                        at com.liulishuo.filedownloader.DownloadTaskHunter.pause(DownloadTaskHunter.java:419)
                                                        at com.liulishuo.filedownloader.DownloadTask.pause(DownloadTask.java:338)
                                                        at com.liulishuo.filedownloader.FileDownloader.pause(FileDownloader.java:249)
                                                        at com.lzj.shanyi.feature.game.download.GameDownloadManager.pause(GameDownloadManager.java:648)
                                                        at com.lzj.shanyi.feature.game.download.subscribe.OnPauseDownloadSubscribe.subscribe(OnPauseDownloadSubscribe.java:31)
                                                        at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
                                                        at io.reactivex.Observable.subscribe(Observable.java:10700)
                                                        at io.reactivex.internal.operators.observable.ObservableSubscribeOn$1.run(ObservableSubscribeOn.java:39)
                                                        at io.reactivex.Scheduler$1.run(Scheduler.java:138)
                                                        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:59)
                                                        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:51)
                                                        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:154)
                                                        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:269)
                                                        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                                                        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                                                        at java.lang.Thread.run(Thread.java:818)
08-23 17:58:14.437 com.lzj.shanyi E/ActivityThread: Failed to find provider info for net.oneplus.odm.provider.PayloadProvider
08-23 17:58:14.437 com.lzj.shanyi E/AbstractTracker: Unknown URL content://net.oneplus.odm.provider.PayloadProvider/OP_payload_upload_table

--- a/library/src/main/java/com/liulishuo/filedownloader/services/FileDownloadThreadPool.java
+++ b/library/src/main/java/com/liulishuo/filedownloader/services/FileDownloadThreadPool.java
@@ -114,7 +114,7 @@ class FileDownloadThreadPool {
         for (int i = 0; i < size; i++) {
             final int key = runnablePool.keyAt(i);
             final DownloadLaunchRunnable runnable = runnablePool.get(key);
-            if (runnable.isAlive()) {
+            if (runnable != null && runnable.isAlive()) {
                 correctedRunnablePool.put(key, runnable);
             }
         }

